src/GraphQL/Server.hs

```diff
@@ -274,22 +312,27 @@ instance forall typeName interfaces fields m.
 -- | Closed type family to enforce the invariant that Union types
 -- contain only Objects.
 type family RunUnionType m (a :: [Type]) :: Type where
-  RunUnionType m (Object typeName interfaces fields:rest) = Handler m (Object typeName interfaces fields) :<|> RunUnionType m rest
-  RunUnionType m '[] = ()
+  RunUnionType m '[Object typeName interfaces fields] = Object typeName interfaces fields
+  RunUnionType m (Object typeName interfaces fields:rest) = Object typeName interfaces fields :<|> RunUnionType m rest
```

Comment:
Ah, this is actually `((Object typeName interfaces fields):rest)` - I can add the parens but I saw you started removing my extra parens that I left in so I removed them as well. Up to you which style we follow I don't mind.